include/SSVOpenHexagon/Utils/Utils.hpp

```diff
@@ -43,19 +43,39 @@ inline void uppercasify(std::string& s)
     return s;
 }
 
-inline float getFontHeight(sf::Text& font)
+[[nodiscard, gnu::pure]] inline float getFontHeight(sf::Text& font)
 {
     font.setString("A");
     return ssvs::getGlobalHeight(font);
 }
 
-inline float getFontHeight(sf::Text& font, const unsigned int charSize)
+[[nodiscard, gnu::pure]] inline float getFontHeight(sf::Text& font, const unsigned int charSize)
 {
     font.setCharacterSize(charSize);
     font.setString("A");
     return ssvs::getGlobalHeight(font);
 }
 
+bool getLinesIntersection(sf::Vector2f& mIntersection,
+    const sf::Vector2f& l1p1, const sf::Vector2f& l1p2,
+    const sf::Vector2f& l2p1, const sf::Vector2f& l2p2);
+
+unsigned int getLineCircleIntersection(sf::Vector2f& i1,
+    sf::Vector2f& i2, const sf::Vector2f& p1,
+    const sf::Vector2f& p2, const float mRadiusSquared);
+
+bool getLineCircleClosestIntersection(
+    sf::Vector2f& mIntersection, const sf::Vector2f& mPos,
+    const sf::Vector2f& p1, const sf::Vector2f& p2,
+    const float mRadiusSquared);
+
+[[nodiscard, gnu::pure]] inline float fastSqrt(const float &n) 
```

Comment:
Did you measure that this is actually faster than `std::sqrt`?

Reply:
It is a well known algorithm that has been used since the Quake engine.
I have not measured myself how it compares to std::sqrt, I will inquire when possible.

Reply:
Run a simple test, this example code returns 0 for the duration of fastSqrt and 1 for std::sqrt
```
#include <iostream>
#include <chrono>
#include <cmath>

inline float fastSqrt(const float &n) 
{
    union{int i; float f;} u;
    u.i = 0x5F375A86 - (*(int*)&n >> 1);
    return (int(3) - n * u.f * u.f) * n * u.f * 0.5f;
}

int main()
{
    const float num = 38725673265436589345.324677;
    
    auto t1 = std::chrono::high_resolution_clock::now();
    const auto f = fastSqrt(num);
    auto t2 = std::chrono::high_resolution_clock::now();
    std::cout << std::chrono::duration_cast<std::chrono::microseconds>( t2 - t1 ).count() << std::endl;
    
    auto t3 = std::chrono::high_resolution_clock::now();
    const auto g = std::sqrt(num);
    auto t4 = std::chrono::high_resolution_clock::now();
    std::cout << std::chrono::duration_cast<std::chrono::microseconds>( t4 - t3 ).count() << std::endl;
    
    return 0;
}
```